TestResultSummaryService/parsers/Parser.js

```diff
@@ -6,17 +6,23 @@ class Parser {
     static canParse() { return false; }
     
     exactJavaVersion(output) {
+        // TODO: Since TRSS is mainly testing on Java performance now, would change javaVersion and jdkDate
+        // to a more generic name later.
+        const nodejsVersionRegex = /(Node Version[\s\S]*Rundate.*)/;
         const javaVersionRegex = /((openjdk|java) version[\s\S]*JCL.*\n|(openjdk|java) version[\s\S]*Server VM.*\n)/;
         const javaBuildDateRegex = /-(20[0-9][0-9][0-9][0-9][0-9][0-9])/;
         let curRegexResult = null;
         let javaVersion, jdkDate;
         if ( ( curRegexResult = javaVersionRegex.exec( output ) ) !== null ) {
             javaVersion = curRegexResult[1];
         }
+        if ( ( curRegexResult = nodejsVersionRegex.exec( output ) ) !== null ) {
+            javaVersion = curRegexResult[1];
+        }

```

Comment:
If a test output has both java version and node version printed, `javaVersion` variable will be set to node version regardless. Also, `exactJavaVersion()` will be modified to check for extra information soon. Please do not nest the node version match in `exactJavaVersion()`. To make code simple and more readable, I would prefer a separate function for the node.